main/src/test/java/org/mobilitydata/gtfsvalidator/util/shape/ShapePointsTest.java

```diff
@@ -0,0 +1,203 @@
+    ShapePoints shapePoints = ShapePoints.fromGtfsShape(gtfsShapes);
+    expect
+        .that(shapePoints.getPoints())
+        .comparingElementsUsing(APPROX_SAME_POINT)
+        .containsExactly(
+            new ShapePoint(0, 0, gtfsShapes.get(0).shapePtLatLon().toPoint()),
+            new ShapePoint(86.294368, 0, gtfsShapes.get(1).shapePtLatLon().toPoint()),
+            new ShapePoint(1104.2600400, 0, gtfsShapes.get(2).shapePtLatLon().toPoint()));
+  }
+
+  @Test
+  public void fromGtfsShape_withUserDistance() {
+    ImmutableList<GtfsShape> gtfsShapes =
+        ImmutableList.of(
+            createGtfsShape(47.3778615, 8.5381339, 0.0),
+            createGtfsShape(47.3771448, 8.5376943, 86.0),
+            createGtfsShape(47.3680534, 8.5361065, 1104.0));
+    ShapePoints shapePoints = ShapePoints.fromGtfsShape(gtfsShapes);
+    expect
+        .that(shapePoints.getPoints())
+        .comparingElementsUsing(APPROX_SAME_POINT)
+        .containsExactly(
+            new ShapePoint(0, 0, gtfsShapes.get(0).shapePtLatLon().toPoint()),
+            new ShapePoint(86.294368, 86.0, gtfsShapes.get(1).shapePtLatLon().toPoint()),
+            new ShapePoint(1104.2600400, 1104.0, gtfsShapes.get(2).shapePtLatLon().toPoint()));
+  }
+
+  @Test
+  public void fromGtfsShape_withDecreasingUserDistance() {
+    ImmutableList<GtfsShape> gtfsShapes =
+        ImmutableList.of(
+            createGtfsShape(47.3778615, 8.5381339, 0.0),
+            createGtfsShape(47.3771448, 8.5376943, 86.0),
+            createGtfsShape(47.3680534, 8.5361065, 70.0));
+    ShapePoints shapePoints = ShapePoints.fromGtfsShape(gtfsShapes);
+    expect
+        .that(shapePoints.getPoints())
+        .comparingElementsUsing(APPROX_SAME_POINT)
+        .containsExactly(
+            new ShapePoint(0, 0, gtfsShapes.get(0).shapePtLatLon().toPoint()),
+            new ShapePoint(86.294368, 86.0, gtfsShapes.get(1).shapePtLatLon().toPoint()),
+            new ShapePoint(1104.2600400, 86.0, gtfsShapes.get(2).shapePtLatLon().toPoint()));
+  }
+
+  @Test
+  public void hasUserDistance() {
+    expect.that(new ShapePoints(ImmutableList.of()).hasUserDistance()).isFalse();
+    expect
+        .that(
+            new ShapePoints(ImmutableList.of(new ShapePoint(0, 0, new S2Point())))
+                .hasUserDistance())
+        .isFalse();
+    expect
+        .that(
+            new ShapePoints(
+                    ImmutableList.of(
+                        new ShapePoint(0, 0, new S2Point()), new ShapePoint(1, 1, new S2Point())))
+                .hasUserDistance())
+        .isTrue();
+    expect
+        .that(
+            new ShapePoints(
+                    ImmutableList.of(
+                        new ShapePoint(0, 0, new S2Point()),
+                        new ShapePoint(1, 1, new S2Point()),
+                        new ShapePoint(2, 0, new S2Point())))
+                .hasUserDistance())
+        .isFalse();
+  }
+
+  private static ShapePoints createShapePoints() {
```

Comment:
Why not storing the object as a static constant instead of creating a new one for each one of the following test cases?

Reply:
This makes more explicit that we create sample objects for each test and then can modify the returned one when necessary. I don't see much difference though.